vtex-apps/store-indexer
# Saves routes in rewriter

# What this PR does?

This PR makes store indexer save internal routes for events related to product, brand and categories. It also saves the product as not Found if it is not active.

Depend on: https://github.com/vtex-apps/store/pull/406

# How to test it 

To test it you can link `vtex.broadcaster-listener`, letting the desired event pass. For example if you want to see the product event just comment this line: 

https://github.com/vtex-apps/broadcaster-listener/blob/57b17470b51ce1f0368365873608e80b4e0cad92/node/middlewares/notify.ts#L85

Now you can do a POST request to `http://broadcaster.vtex.aws-us-east-1.vtex.io/<account>/<workspace>/notify` with the following body: 

```
{
	"HasStockKeepingUnitModified": "true",
	"IdSku":  <sku-id>
}
``` 
This request will make store-indexer re-index the entities related to the sku. You can check if the route was saved correctly by accessing `vtex.rewriter` graphql api.
